### PR TITLE
Add test for bridge flags for enabling and disabling flooding on ports.

### PR DESCRIPTION
**Pull request by cappe987**
Tests that flooding behaves as intended when both enabled and disabled, for both unicast and multicast.

Minor correction of a comment.

### CODE REVIEW

File: lib.sh [hunk 1]
```diff
@@ -111,7 +111,7 @@ ipaddr()
 #   d: local dest addr from arg
 #   g: global dest addr from arg (multicast)
 #   i: source addr from interface arg
-#   i: dest addr from interface arg
+#   I: dest addr from interface arg
```
Comment (wkz) [inline lib.sh:114]: Good catch! Please split it out in a separate commit :+1:

File: suite/20-basic.sh [hunk 2]
```diff
@@ -91,3 +91,49 @@ basic_learning_port()
...
+		eth -i $h1 | { cat; echo from $h1; } | inject $h1
+		report $h2 | grep -q "from $h1" && fail
+		report $h3 | grep -q "from $h1" || fail
+
+		step "Verify multicast flood on"
+		# Craft multicast address
+		mcast_da="\001\000\136\252\252\252" # 01:00:5e:aa:aa:aa
+		mcast_sa="$(addrfmt $(rndaddr))"
+		et="$(be16 0xbbbb)"
+		mcast_target="${mcast_da}${mcast_sa}${et}"
```
Comment (wkz) [inline suite/20-basic.sh:122]: Any particular reason for generating a DA in the IPv4 multicast range? I think the SA could also just be `h1`s interface address. If yes, then the next comment applies :smile:

File: suite/20-basic.sh [hunk 3]
```diff
@@ -91,3 +91,49 @@ basic_learning_port()
...
+
+		bridge link set dev $b2 mcast_flood on
+		capture $h2 $h3
+		printf "$mcast_target" | { cat; echo from $h1; } | inject $h1
```
Comment (wkz) [inline suite/20-basic.sh:126]: So I think this could just be:

`eth -g 1 -i $h1 |  { cat; echo from $h1; } | inject $h1`

File: suite/20-basic.sh [hunk 4]
```diff
@@ -91,3 +91,49 @@ basic_learning_port()
...
+		report $h2 | grep -q "from $h1" || fail
+		report $h3 | grep -q "from $h1" || fail
+
+		step "Verify multicast flood off"
+		bridge link set dev $b2 mcast_flood off
+		capture $h2 $h3
+		printf $mcast_target | { cat; echo from $h1; } | inject $h1
```
Comment (wkz) [inline suite/20-basic.sh:133]: ...and naturally the same applies here.

**Review by wkz — COMMENTED** [review]
Great stuff! :1st_place_medal: 

Just want to see if we can avoid the escape-character-fest around the multicast part.

**Comment by cappe987** [comment]
No particular reason for the IPv4 multicast. I needed a multicast address and I made one :). Now I've switched to your suggestion. Also split up the correction into another commit.
